(Gradle) Running tests with agent fails with "Missing property userCodeFilterPath in agent conditional configuration"

**Describe the bug**
Running `./gradlew -Pagent nativeTest` as per [documentation](https://graalvm.github.io/native-build-tools/latest/gradle-plugin.html#agent-support) in a project with no configuration (except adding the plugin), fails with

```
Caused by: org.gradle.api.internal.provider.AbstractProperty$PropertyQueryException: Failed to calculate the value of property 'agentOptions'.
        at org.gradle.api.internal.provider.AbstractProperty.finalizeNow(AbstractProperty.java:248)
        at org.gradle.api.internal.provider.AbstractProperty.beforeRead(AbstractProperty.java:239)
        at org.gradle.api.internal.provider.AbstractProperty.calculateOwnValue(AbstractProperty.java:135)
        at org.gradle.api.internal.provider.AbstractMinimalProvider.getOrNull(AbstractMinimalProvider.java:105)
        at org.gradle.api.internal.provider.ProviderResolutionStrategy$1.resolve(ProviderResolutionStrategy.java:27)
        at org.gradle.util.internal.DeferredUtil.unpack(DeferredUtil.java:59)
        at org.gradle.util.internal.DeferredUtil.unpackOrNull(DeferredUtil.java:49)
        at org.gradle.api.internal.tasks.properties.InputParameterUtils.prepareInputParameterValue(InputParameterUtils.java:38)
        at org.gradle.api.internal.tasks.properties.InputParameterUtils.prepareInputParameterValue(InputParameterUtils.java:30)
        ... 72 more
Caused by: org.gradle.api.GradleException: Missing property userCodeFilterPath in agent conditional configuration
        at org.graalvm.buildtools.gradle.internal.agent.AgentConfigurationFactory.lambda$getAgentConfiguration$653f1fdd$1(AgentConfigurationFactory.java:81)
        at org.gradle.api.internal.provider.ValueSupplier$Present.transform(ValueSupplier.java:541)
        at org.gradle.api.internal.provider.TransformBackedProvider.mapValue(TransformBackedProvider.java:91)
        at org.gradle.api.internal.provider.TransformBackedProvider.calculateOwnValue(TransformBackedProvider.java:83)

```

env:
* Gradle: 8.5 (latest)
* 'org.graalvm.buildtools.native' version '0.9.28' (latest)
* GraalVM: 21.0.1 CE (latest)
* Reproducer: branch, https://github.com/abelsromero/java-parallels/tree/add-native-java-compatibility

**Make sure that you have read [the documentation](https://graalvm.github.io/native-build-tools) and that you are using the latest plugin version.**

**To Reproduce**

```
git clone -b add-native-java-compatibility git@github.com:abelsromero/java-parallels.git
cd java-parallels
./gradlew -Pagent nativeTest
```

**Expected behavior**
It should run tests and provide auto-generated metadata 🙏 

**Logs**


**System Info (please complete the following information):**
 - OS: MacOs
 - GraalVM Version: 21.0.1 CE
 - Java Version: 21
 - Plugin version: 'org.graalvm.buildtools.native' version '0.9.28'

**Additional context**

--- a/docs/src/docs/asciidoc/gradle-plugin.adoc
+++ b/docs/src/docs/asciidoc/gradle-plugin.adoc
@@ -248,6 +248,10 @@ The agent can run in multiple modes that dictate how the metadata is collected a
 * `tasksToInstrumentPredicate` - By default, when `-Pagent` is specified, all tasks extending `JavaForkOptions` are instrumented.
 This can be limited to only specific tasks that match this predicate.
 
+When you pass `-Pagent` without an explicit mode, the plugin uses `defaultMode`.
+The default `defaultMode` is `standard`, except projects applying Gradle's `java-library` plugin default to `conditional`.
+Because `conditional` mode requires `userCodeFilterPath`, library projects should either configure a conditional user-code filter or run with an explicit mode, for example `-Pagent=standard`.
+
 [source,groovy,subs="verbatim,attributes", role="multi-language-sample"]
 ----
 agent {

--- a/native-gradle-plugin/docs/functional/tracing-agent.md
+++ b/native-gradle-plugin/docs/functional/tracing-agent.md
@@ -7,7 +7,8 @@ The Gradle plugin exposes the shared tracing-agent workflow from
 
 The agent can be enabled by DSL configuration or with the `-Pagent` Gradle property. When the
 property names a mode, that mode must override the configured default mode for the instrumented
-run.
+run. The default mode convention is `standard`, except projects that apply Gradle's
+`java-library` plugin use `conditional` as the default convention.
 
 ## 2. Instrumented tasks
 

